scripts/CsvCaseDataUpdater.java

```diff
@@ -0,0 +1,253 @@
+        return (file.exists() && !file.isDirectory());
+    }
+
+    /**
+     * Reads the CSV and parses into a list of CaseData's, each representing a change
+     * to a specific case.
+     *
+     * @param csvFile the CSV file that will be read
+     * @return a list of CaseData's containing the required changes
+     * @throws IOException if the reading of the file doesn't succeed.
+     */
+    private static List<CaseData> readFile(final File csvFile) throws IOException {
+        List<CaseData> caseData = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
+            final String[] headers = parseHeaders(br.readLine());
+
+            String line = br.readLine();
+            while (line != null) {
+                String[] values = line.split(",");
+
+                if (values.length >= 1) {
+                    // [0] is the case UUID
+                    CaseData data = new CaseData(values[0]);
+
+                    for(int i = 1; i < values.length; i++) {
+                        String value = values[i];
+                        if (!value.equals("")) {
+                            // Header and the value will correspond to the same index
+                            data.addToMap(headers[i], value);
+                        }
+                    }
+
+                    caseData.add(data);
+                }
+                line = br.readLine();
+            }
+        }
+
+        return caseData;
+    }
+
+    /**
+     * Returns the headers of a CSV file, used for calculating what fields should be updated.
+     * Splits by the comma delimiter.
+     *
+     * @param headerLine the first line in the CSV that represents the fields
+     * @return String[] of the headers
+     */
+    private static String[] parseHeaders(final String headerLine) {
+        if (headerLine == null) {
+            throw new IllegalArgumentException("Inputted header line cannot be null.");
+        }
+
+        String[] headers = headerLine.split(",");
+        if (headers.length == 0) {
+            throw new IllegalArgumentException("No headers have been found, please ensure these are in the CSV.");
+        }
+
+        return headers;
+    }
+
+    /**
+     * Updates the inputted cases with the new values for the values supplied.
+     *
+     * @param allCaseData a list of CaseData objects that contain the fields that
+     *                    should be changed.
+     */
+    private static void updateCases(List<CaseData> allCaseData) {
+        final HttpClient client = HttpClient.newHttpClient();
+
+        for (CaseData caseData: allCaseData) {
+            final List<HttpRequest> requests = generateUpdateRequestForCase(caseData);
+
+            // Loop round each request and send to the casework service
+            requests.forEach(request -> {
+                try {
+                    final HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
```

Comment:
Could this log the changes made?
I'm not sure if possible but somehow log to Kibana?

Reply:
Added logging into the program, for all of the requests and also the success messages. As we throw an exception on a invalid request/ non-200 result, that already acts as logging there.